packages/material-ui/src/InputBase/InputBase.test.js

```diff
@@ -159,6 +159,42 @@ describe('<InputBase />', () => {
     });
   });
 
+  // Note the initial callback when
+  // uncontrolled only fires for a full mount
+  describe('uncontrolled', () => {
+    it('should fire the onFilled callback when initially provided a value', () => {
+      function FilledState(props) {
+        const { filled } = useFormControl();
+        return <span {...props}>filled: {String(filled)}</span>;
+      }
+
+      function TestUncontrolled() {
+        const refs = React.useRef({});
+
+        return (
+          <FormControl>
+            <FilledState data-testid="filled" />
+            <InputBase
+              name="test"
+              inputRef={ref => {
+                if (ref) {
+                  refs.current[ref.name] = ref;
+                  refs.current[ref.name].value = 'test@gmail.com';
```

Comment:
We already have a way of specifying an initial value on an uncontrolled input component: `<input defaultValue={initialValue} />`. I'd hope we already have a passing test for that behavior. I don't think we should support side-effects outside of react.